Fix bug where SubSlot was detaching unnecessarily

TW80000 opened the pull request:
SubSlots were detaching whenever the class of the object in the parent
slot changed. The reason for this, according to a comment in the code,
is that the new class will have "different SubSlots", which I assume
means different axioms.

An unfortunate side effect of this "feature" is that when the class of
the object in the parent slot changes to another class with the same
axiom, the SubSlot detaches anyway. It would be preferable, in my
opinion, if the SubSlot stays alive as long as the new class has the
same axiom the subslot was set up for. That's what this commit
implements.

Commit:
Fix bug where SubSlot was detaching unnecessarily

SubSlots were detaching whenever the class of the object in the parent
slot changed. The reason for this, according to a comment in the code,
is that the new class will have "different SubSlots", which I assume
means different axioms.

An unfortunate side effect of this "feature" is that when the class of
the object in the parent slot changes to another class with the same
axiom, the SubSlot detaches anyway. It would be preferable, in my
opinion, if the SubSlot stays alive as long as the new class has the
same axiom the subslot was set up for. That's what this commit
implements.

kgrgreer commented on src/foam/core/Slot.js:
```diff
@@ -334,10 +334,14 @@ foam.CLASS({
-      if ( o && ( this.of !== o.cls_.id || o.cls_.getAxiomByName(this.name) == null ) ) {
+      // If the new class has the same axiom as the old class, then we keep this
+      // SubSlot attached instead of detaching it.
+      if ( o && ( this.of !== o.cls_.id && o.cls_.getAxiomByName(this.name) == null ) ) {
```
Is this part redundant: this.of !== o.cls_.id &&

TW80000 replied:
Resolved by 7c8235d

Commit:
Don't track information we don't need

Checking the axiom is sufficient. We don't actually need to check the
class of the object in the parent slot.
